C7/Util.cs

```diff
@@ -5,48 +5,65 @@
 using Godot;
 using ConvertCiv3Media;
 
-public class Util
-{
+public class Util {
 	static public string Civ3Root = GetCiv3Path();
 	public class Civ3FileDialog : FileDialog
 	// Use this instead of a scene-based FileDialog to avoid it saving the local dev's last browsed folder in the repo
 	// While instantiated it will return to the last-accessed folder when reopened
 	{
 		public string RelPath= "";
-		public Civ3FileDialog(FileDialog.ModeEnum mode = FileDialog.ModeEnum.OpenFile)
-		{
+		public Civ3FileDialog(FileDialog.ModeEnum mode = FileDialog.ModeEnum.OpenFile) {
 			Mode = mode;
 		}
-		public override void _Ready()
-		{
+		public override void _Ready() {
 			Access = AccessEnum.Filesystem;
-			CurrentDir = Util.GetCiv3Path() + "/" + RelPath;
+			CurrentDir = Civ3Root + "/" + RelPath;
 			Resizable = true;
 			MarginRight = 550;
 			MarginBottom = 750;
 			base._Ready();
 		}
 
 	}
-	static public string GetCiv3Path()
-	{
+
+	static private string SteamCommonDir() {
+		string home = System.Environment.GetFolderPath(System.Environment.SpecialFolder.Personal);
+		if (home == null) { return null; }
+		return System.IO.Path.Combine(home, "Library/Application Support/Steam/steamapps/common");
+	}
+
+	static private bool FolderIsCiv3(System.IO.DirectoryInfo di) {
+		return di.EnumerateFiles().Any(f => f.Name == "civ3id.mb");
+	}
+
+	static public string GetCiv3Path() {
 		// Use CIV3_HOME env var if present
 		string path = System.Environment.GetEnvironmentVariable("CIV3_HOME");
-		if (path != null) return path;
-
-		// Look up in Windows registry if present
-		path = Civ3PathFromRegistry();
-		if (path != null) return path;
+		if (path != null) { return path; }
+
+		if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows)) {
+			// Look up in Windows registry if present
+			path = Civ3PathFromRegistry();
+			if (path != null) { return path; }
+		} else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX) || RuntimeInformation.IsOSPlatform(OSPlatform.Linux)) {
+			// Check for a civ3 folder in steamapps/common
+			// TODO: should this method only be used at dev time?
+			var root = new System.IO.DirectoryInfo(Util.SteamCommonDir());
```

Comment:
Minor: We're trying out best to always specify the type, to make it easier to read going forward (in this case that would be DirectoryInfo).

My thought would be we should leave this check on all the time as it will make it convenient for new users.